Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+                                                                                                @{N='isMinTLSVersionSetOnCustomDomain';E={$frontDoorsEndpointWithoutReqMinTLSVersion.isMinTLSVersionSetOnCustomDomain}}
+     
+                                                                                
+            }
+        }
+    
+    $totalFrontDoorWithoutReqMinTLSVersion = ($frontDoorWithoutReqMinTLSVersion | Measure-Object).Count
+
+    if ($totalFrontDoorWithoutReqMinTLSVersion -eq 0)
+    {
+        Write-Host "No Front Door(s) found having minimum TLS version less than required minimum TLS version. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)
+        if($AutoRemediation -and ($frontDoorsWithReqMinTLSVersion|Measure-Object).Count -gt 0) 
+        {
+            $logFile = "LogFiles\"+ $($TimeStamp) + "\log_" + $($SubscriptionId) +".json"
+            $log =  Get-content -Raw -path $logFile | ConvertFrom-Json
+            foreach($logControl in $log.ControlList){
+                if($logControl.ControlId -eq $controlIds){
+                    $logControl.RemediatedResources=$logRemediatedResources
+                    $logControl.SkippedResources=$logSkippedResources
+                }
+            }
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }
+        return
+    }
+
+    Write-Host "Found [$($totalFrontDoorWithoutReqMinTLSVersion)] out of [$($totalFrontDoors)] Front Door(s) having minimum TLS version less than required minimum TLS version." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetMinTLSVersionForCdnFrontDoors"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    if (-not $DryRun)
+    {
+        if(-not $SkipBackup)
+        {
+            Write-Host "Backing up Front Doors details to [$($backupFolderPath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            $backupFile = "$($backupFolderPath)\CdnFrontDoorsWithoutReqMinTLSVersion.csv"
+            $frontDoorWithoutReqMinTLSVersion | Export-CSV -Path $backupFile -NoTypeInformation
+            Write-Host "Front Doors details have been backed up to [$($backupFile)]." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        Write-Host "Minimum required TLS Version will be set on all custom domains of front doors." -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
NIT: Can we also say to 1.2

Reply:
Added required minimun tls version there.